library/src/armory/logging.py

```diff
@@ -0,0 +1,61 @@
+"""Utilities to autoconfigure logging for Armory."""
+
+import logging
+import os
+import sys
+
+
+def configure_logging(
+    armory_level: int = logging.INFO,
+    datasets_level: int = logging.WARNING,
+    lightning_level: int = logging.WARNING,
+    mlflow_level: int = logging.WARNING,
+    transformers_level: int = logging.WARNING,
+    enable_basic_config: bool = True,
+    enable_progress_bars: bool = sys.stdout.isatty(),
+    **kwargs,
+):
```

Comment:
It makes sense for Armory Library to configure logging for sub-components like `Lightning` and `MLFlow`. What about optional packages that may be used in an evaluation such as `datasets` and `transformers`? For instance, Armory currently relies on the Adversarial Robustness Toolbox, but `configure_logging` does not include ART.

Reply:
I included configuration of the optional packages (datasets, transformers) because they're non-trivial (i.e., more than `logging.getLogger(...).setLevel(...)`) and very likely to be used with Armory. I made sure that their imports are wrapped in try/except, so if you aren't using them and they're not installed, that this function will still work.

We could just as easily set the ART logging level, but I haven't found that ART produces much in the way of logging. Lightning and MLFlow were putting out info-level logs that weren't really helpful. At that point, whether to set the log level for ART feels no different than Captum, XAITK-salience, etc. In that, you have to draw the line somewhere. Since ART's logs aren't problematic, I had no reason to set its logging level.